lib/charms/opensearch/v0/opensearch_plugin_manager.py

```diff
@@ -0,0 +1,65 @@
+# Copyright 2023 Canonical Ltd.
+# See LICENSE file for licensing details.
+
+"""Implements the plugin manager class."""
+
+from typing import Dict, List
+
+from charms.opensearch.v0.opensearch_plugins import OpenSearchPlugin
+from ops.framework import Object
+from ops.model import ActiveStatus, StatusBase
+
+# The unique Charmhub library identifier, never change it
+LIBID = "da838485175f47dbbbb83d76c07cab4c"
+
+# Increment this major API version when introducing breaking changes
+LIBAPI = 0
+
+# Increment this PATCH version before using `charmcraft publish-lib` or reset
+# to 0 if you are raising the major API version
+LIBPATCH = 1
+
+
+OpenSearchPluginsAvailable = {}
+
+
+class OpenSearchPluginManager:
+    """Manages the currently enabled plugins."""
+
+    def __init__(self, charm: Object):
+        self._charm = charm
+
+    @property
+    def plugins(self) -> Dict[str, OpenSearchPlugin]:
+        """Returns dict of installed plugins."""
+        return {
+            key: plugin_data["class"](key, self._charm)
+            for key, plugin_data in OpenSearchPluginsAvailable.items()
+        }
+
+    def plugin_map_config_name_to_class(self) -> Dict[str, OpenSearchPlugin]:
+        """Returns dict of plugins installed either via config or relation.
```

Comment:
This method seems to be redundant: `plugins` should be enough. (not sure I understand "installed via relation"?)

Reply:
@Mehdi-Bendriss that is the inverted map, and I'm using both, actually